src/verify/index.ts

```diff
@@ -10,7 +21,7 @@ export function verify(
   secret: string,
   eventPayload: object,
   signature: string
-): boolean {
+): eventPayload is GithubEvent {
```

Comment:
could you explain how that helps exactly? I'm not familiar with that notion

Reply:
No problem - type guards allow you to define functions that narrow types based on if they return true or false. We actually now also have assertion guards which are the same except the function throws (they landed a few months ago, and very cool).

So by making this a type-guard, you can do this:

```
const signature = event.headers['X-Hub-Signature'];
const { GITHUB_WEBHOOK_SECRET } = process.env;
const body = JSON.parse(event.body ?? '{}') as object;

if (verify(GITHUB_WEBHOOK_SECRET, body, signature)) {
  console.log('Got an event named', body.name, 'from GitHub');
}
```

In the above, because `verify` is a type guard, TypeScript knows that in the scope of the if statement `body` *has* to be of type `GithubEvent` when that code is running since it only runs when its `true`.

`Array.isArray` is an example of a built-in type guard - that function is defined as `isArray<T>(arg: T | {}): arg is T[]`.

(it's actually a bit bigger, as it has a conditional check to say `"arg is (if T is readonly ? readonly T[] : T[])"`, but I've removed that to keep things simple)

Reply:
[Here's the code in action, vs the current method](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgUQG4FMB2MAqBPMdAZwAUBDPAGwjIBMAaODKYAMzzgF85WoIQ4AcgACEAMYwIAa2AwA9AHd0AIwAWEaUUEBuAFC6YBdHADqK9dLRYYROAF4UADzGUArrXQAeXXDhT0eBCsKBjY+ISkFNR09D5wAAYAJAhEMCyYAOacAHTJqelZ8XAAPnAAROhQfFBEZSXlAFRlugB8egZGcADisqquylbYnjgAcmQgxuiOMFi0tmZqGlKDNvam5ksrRC1rAPIgst6+K+HE5FQ0tADao+PoALqxvmWYd807AGSIcK8TAFxwW4TLjtMQQTCpOAZdAwACClAy0F6AgcAAoiMAMq8YK4oOgAflgJkAJT2HYIOJ4nFQTBwDFYsjU9DZVJkWBEEy9VFlIiqMgAJgArAA2OxlUkAfnKvIFIrqAJ5fIAjGU9Jx2h4XGzjKxXJgJMBwUxKmw8PzUXEiOgxFSCWkiRknnB0KEYOdorQARBlAAra0wJ307G4-F0+2ZXTEgEu6zuy5wYC2HowPoDV2g8GQ1i0tHKCC0PBe33+0l2FoU3xsVHMU2owQgPBWm0wwSMPMFxj1xuY4N4wTE0kV3xwMEQiCUZnUDJ1roQeBkWkx7A-O60Vtwdt4bK-dCd3j8bqyAAS-X7el8nH0ldY1ZN7HNXabVPXm87DaDjJD-cHcV8o6I46ThA06CLO86Lq6K4TGubb5luO57nwAjJieyhnnEl6XkAA).

You'll see that the first use of `body.name` has an error, whereas the second doesn't:

![image](https://user-images.githubusercontent.com/3151613/101266146-12dacb80-37b1-11eb-9a63-d9289096a48a.png)

Reply:
Perfect, I got it now, makes a lot of sense. Thank you for the thorough explanation!